src/build-data/botan-config.cmake.in

```diff
@@ -0,0 +1,42 @@
+set(_Botan_mod_list "%{mod_list}")
+string(REGEX MATCHALL "[A-Za-z_0-9]+" _Botan_supported_components ${_Botan_mod_list})
+
+foreach(_comp ${Botan_FIND_COMPONENTS})
+  if (NOT _comp IN_LIST _Botan_supported_components)
+    set(Botan_FOUND False)
```

Comment:
You need to check if the component is optional before failing the entire package. This should be `Botan_${_comp]_FOUND`